doc/api/stream.md

```diff
@@ -1722,6 +1722,141 @@ const cleanup = finished(rs, (err) => {
   // ...
 });
 ```
+#### Class: `stream.Body`
+<!-- YAML
+added: REPLACEME
+-->
+
+<!--type=class-->
+
+`new stream.Body` can be used to seamlessly convert between different types of
+read/write stream interfaces, including async functions and iterables.
+
+* `AsyncIterable` converts into a readable. Cannot yield
+  `null`.
+* `AsyncGeneratorFunction` converts into a readable/writable.
+  Must take a source `AsyncIterable` as first parameter. Cannot yield
+  `null`.
+* `AsyncFunction` converts into a writable. Must return
+  either `null` or `undefined`.
+
+```mjs
+import { Body, compose } from 'stream';
+import { finished } from 'stream/promises';
+// Convert AsyncIterable into readable Duplex.
+const s1 = new Body(async function*() {
+  yield 'Hello';
+  yield 'World';
+}());
+// Convert AsyncGenerator into transform Duplex.
+const s2 = new Body(async function*(source) {
+  for await (const chunk of source) {
+    yield String(chunk).toUpperCase();
+  }
+});
+let res = '';
+// Convert AsyncFunction into writable Duplex.
+const s3 = new Body(async function(source) {
+  for await (const chunk of source) {
+    res += chunk;
+  }
+});
+await finished(compose(s1, s2, s3).toNodeStream());
+console.log(res); // prints 'HELLOWORLD'
+```
+
+### `body.arrayBuffer()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
+
+Returns a promise that fulfills with an {ArrayBuffer} containing a copy of
+the body data.
+
+### `body.blob()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
+
+Returns a promise that fulfills with an {Blob} containing a copy of the body data.
+
+### `body.buffer()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
+
+Returns a promise that fulfills with an {Buffer} containing a copy of the body data.
+
+### `body.nodeStream()`
```

Comment:
Since we already have the node-to-web adapters, and your new `compose` function hopefully landing soon, I don't think we need these. Let's keep this focused on the accumulator funtions.